tests/dogfood.rs

```diff
@@ -179,8 +179,39 @@ fn dogfood_subprojects() {
 #[ignore]
 #[cfg(feature = "metadata-collector-lint")]
 fn run_metadata_collection_lint() {
+    use std::fs::File;
+    use std::time::SystemTime;
+
+    // Setup for validation
+    let metadata_output_path = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("util/gh-pages/metadata_collection.json");
+    let start_time = SystemTime::now();
+
+    // Run collection as is
     std::env::set_var("ENABLE_METADATA_COLLECTION", "1");
     run_clippy_for_project("clippy_lints");
+
+    // Check if cargo caching got in the way
+    if let Ok(file) = File::open(metadata_output_path) {
+        if let Ok(metadata) = file.metadata() {
+            if let Ok(last_modification) = metadata.modified() {
+                if last_modification > start_time {
```

Comment:
Is this even possible? This only happens if someone modifies the file between 6 lines before and this line.

Reply:
This can happen if you don't remove the `metadata_collection.json` file before running this test. It would also be an option to delete it beforehand and simply check if a new one was created. This just removes the deletion step.

Reply:
I meant the other way around: If you don't delete the file, it exists (step 1 in the timeline). Then you run the metadata collection (step 2 in the timeline). So `start_time > last_modified` in this scenario.  The only way that start time is smaller than the last modification is, if you modify it during the execution of those 6 LoC. I don't think a human is capable of such a precise timing.

Reply:
Ahh, the six previous lines also include run_clippy_for_project("clippy_lints"); which runs the metadata collector itself. This makes it humanly possible (if no cache is used) but most importantly makes it possible for Clippy to rewrite the file. And this time check is targeted towards manipulation from our monster.

The test basically only touches clippy_lints/src/lib.rs if cargo actually used caching.

Reply:
> Ahh, the six previous lines also include `run_clippy_for_project("clippy_lints");` 

Oh, I missed that. This should be fine then.